packages/astro/src/core/config.ts

```diff
@@ -481,16 +477,35 @@ interface TryLoadConfigResult {
 async function tryLoadConfig(
 	configOptions: LoadConfigOptions,
 	flags: CLIFlags,
-	userConfigPath: string | undefined,
 	root: string
 ): Promise<TryLoadConfigResult | undefined> {
+	let configPath = (await resolveConfigURL({ cwd: configOptions.cwd, flags: configOptions.flags }))
+		?.pathname;
+	if (!configPath) return undefined;
+	let finallyCleanup = async () => {};
 	try {
-		// Automatically load config file using Proload
-		// If `userConfigPath` is `undefined`, Proload will search for `astro.config.[cm]?[jt]s`
+		if (configOptions.invalidateWithCache) {
+			// Hack: Write config to temporary file at project root
+			// This invalidates and reloads file contents when using ESM imports or "resolve"
+			const tempConfigPath = vite.normalizePath(
```

Comment:
does this only happen on reloads? If so it would be nice to comment that.

Reply:
Correct, only reloads! May change the variable name to `isConfigReload` instead of `invalidateWithCache`, to show the context you would use it